.github/workflows/linearizability-template.yaml

```diff
@@ -0,0 +1,48 @@
+name: Reusable Linearizability Workflow
+on:
+  workflow_call:
+    inputs:
+      ref:
+        required: true
+        type: string
+      count:
+        required: true
+        type: number
+      testTimeout:
+        required: false
+        type: string
+        default: '30m'
+permissions: read-all
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@ac593985615ec2ede58e132d2e21d2b1cbd6127c # v3.3.0
+    - uses: actions/setup-go@6edd4406fa81c3da01a34fa6f6343087c207a568 # v3.5.0
+      with:
+        go-version: '1.19.5'
+    - name: build
+      env:
+        GITHUB_REF: ${{ inputs.ref }}
+      run: |
+        case "${GITHUB_REF}" in
+          release-3.5)
+            make build-failpoints-release-3.5
+            ./bin/etcd --version
+            ;;
+          release-3.4)
+            make build-failpoints-release-3.4
+            ./bin/etcd --version
+            ;;
+          *)
+            make gofail-enable
+            make build
+            ;;
```

Comment:
```suggestion
          main)
            make gofail-enable
            make build
            ./bin/etcd --version
            ;;
          *)
            echo "Invalid GITHUB_REF: ${GITHUB_REF}"
            exit 1
            ;;            
```

Reply:
That's not correct because on PR we get `GITHUB_REF=refs/pull/<pr_number>/merge`